Packs/CommonScripts/Scripts/GetEndpointData/GetEndpointData.py

```diff
@@ -0,0 +1,719 @@
+
+    This function takes a list of endpoint dictionaries and combines them into a single dictionary.
+    If conflicts occur for keys other than 'Hostname', the values are combined into a list.
+    For 'Hostname', if a conflict is detected, an error is logged.
+
+    Args:
+        endpoints (list[dict[str, dict[str, Any]]]): A list of endpoint dictionaries to merge.
+
+    Returns:
+        dict[str, Any]: A merged dictionary containing all endpoint information.
+
+    Note:
+        - Conflicts for 'Hostname' are treated as errors and logged.
+        - For other keys, conflicting values are combined into a list.
+    """
+    demisto.debug(f'merging endpoints with {endpoints=}')
+    merged_endpoint: dict[str, Any] = {}
+    for endpoint in endpoints:
+        demisto.debug(f'current endpoint: {endpoint}')
+        for key, value in endpoint.items():
+            # If a different hostname was somehow returned by a vendor
+            if key == 'Hostname' and key in merged_endpoint and value['Value'] != merged_endpoint[key]['Value']:
+                demisto.error(f"Conflict detected for 'Hostname'. Conflicting dictionaries: {merged_endpoint[key]}, {value}")
+            # For other keys, add to list if conflict exists
+            elif key in merged_endpoint:
+                if isinstance(merged_endpoint[key], list):
+                    merged_endpoint[key].append(value)
+                else:
+                    merged_endpoint[key] = [merged_endpoint[key], value]
+            else:
+                merged_endpoint[key] = value
+
+    return merged_endpoint
+
+
+def extract_context_outputs(output_keys: list[str], raw_context: list[dict[str, Any]]) -> list[list[dict[str, Any]]]:
+    demisto.debug(f'extracting context outputs from {output_keys=} with {raw_context=}')
+    normalized_data = []
+
+    for context in raw_context:
+        # Convert each key's data to a list using to_list
+        lists_of_objects = [to_list(get_outputs(key, context)) for key in output_keys]
+
+        # Use zip to group corresponding elements together
+        for grouped_objects in zip(*lists_of_objects):
+            demisto.debug(f'{grouped_objects =}')
+            normalized_data.append(list(grouped_objects))
+
+    return normalized_data
+
+
+def get_raw_endpoints(context_outputs: list[list[dict[str, Any]]]) -> list[dict[str, Any]]:
+    raw_endpoints = []
+    for data_list in context_outputs:
+        raw_endpoint = {}
+        for raw_data in data_list:
+            raw_endpoint.update(raw_data)
+        raw_endpoints.append(raw_endpoint)
+    return raw_endpoints
+
+
+def create_endpoints(raw_endpoints: list[dict[str, Any]], output_mapping: dict | Callable, brand: str) -> list[dict[str, Any]]:
+    endpoints = []
+    for raw_endpoint in raw_endpoints:
+        if isinstance(output_mapping, dict):
+            output_map = output_mapping
+        else:
+            output_map = output_mapping(raw_endpoint)
```

Comment:
```suggestion
        output_map = output_mapping(raw_endpoint) if not isinstance(output_mapping, dict) else output_mapping

```

Reply:
what about `output_map = output_mapping(raw_endpoint) if isinstance(output_mapping, Callable) else output_mapping`. Makes it a one liner like you suggested, plus, making sure it is callable